### PR TITLE
Implement the breeze add-back-references command

### PR DESCRIPTION
**Pull request by amoghrajesh**
As the final leg of the feature for adding a new breeze command to publish docs, this PR adds support to move the add-back-references to breeze under release-management. Reference: https://github.com/apache/airflow/issues/32491#issuecomment-1632381565

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: BREEZE.rst [hunk 1]
```diff
@@ -2020,6 +2020,38 @@ Those are all available flags of ``release-management publish-docs`` command:
...
+use the ``release-management add-back-references`` command. This is important to support backward compatibility
+the airflow documentation.
+
+.. code-block:: bash
+
+     breeze release-management add-back-references
+
+.. code-block:: bash
+
+     release-management add-back-references --airflow-site-directory
```
Comment (potiuk) [inline BREEZE.rst]:
```suggestion
     release-management add-back-references --airflow-site-directory DIRECTORY
```

**Review by potiuk — COMMENTED** [review]
Looks cool. Few NITs

**Comment by amoghrajesh** [comment]
@potiuk I think we are green on this PR. Can you take a pass at it when you have some time?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool. Now you  need to update release instructions and remove the old script from airlfow-sites as a follow up.

**Pull request merged**